packages/grid/_modules_/grid/hooks/features/treeData/gridTreeDataUtils.ts

```diff
@@ -0,0 +1,88 @@
+import { GridFilterState } from '../filter';
+import { GridRowId, GridRowTreeConfig, GridRowTreeNodeConfig } from '../../../models';
+
+interface FilterRowTreeFromTreeDataParams {
+  rowTree: GridRowTreeConfig;
+  disableChildrenFiltering: boolean;
+  isRowMatchingFilters: ((rowId: GridRowId) => boolean) | null;
+}
+
+/**
+ * A node is visible if one of the following criteria is met:
+ * - One of its children is passing the filter
+ * - It is passing the filter
+ */
+export const filterRowTreeFromTreeData = (
```

Comment:
Copy / paste from `useGridFilter`
The filtering behavior between `useGridTreeData` and `useGridGroupingColumns` is to different to have the same method.